test/model/src/core/data_value.cpp

```diff
@@ -41,6 +45,73 @@ namespace model {
  */
 const data_value NONE = data_value::create_none();
 
+/*
+ * item_to_string --
+ *     Unpack a WiredTiger byte buffer into a string. This works because C++ strings allow NULL
+ *     bytes to be included within the string, but this is arguably not the best solution.
+ */
+inline static std::string
+item_to_string(const WT_ITEM &item)
```

Comment:
Yeah, I felt like that `std::string` would just do, as it can support basically arbitrary bytes, and it was convenient because I didn't have to add support for a yet another type to the `data_value`'s `std::variant`. But you are right that an actual byte vector would be cleaner—so I just did that.